public/js/index.js

```diff
@@ -0,0 +1,35 @@
+/**/
+$(function () {
+	var socket = io();
+	var user = new User();
+
+	$('form').submit(function() {
+		var msg =  $('#m').val();
+		var userName = $('#name' ).val();
+
+		if (userName.length === 0) {
+			alert('Please identify yourself');
+			return false;
+		}
+		if (user.name !== userName)
+			user = new User(userName);
```

Comment:
In the background you are createing new UI instance per userName change, is it necessary?